### ISSUE TITLE
Request generator for hotelReservation is not working.

### ISSUE THREAD

**Issue by mrbeann** [delimitrou/DeathStarBench#32]
I deploy the hotelReservation and request it using the command `./wrk -D exp -t 4 -c 4 -d 10 -L -s ./wrk2_lua_scripts/mixed-workload_type_1.lua http://localhost:5000 -R 300`, but the result shows there are no successful request. Is there any problem?

```
#[Mean    =        0.758, StdDeviation   =        2.194]
#[Max     =       39.872, Total count    =         1493]
#[Buckets =           27, SubBuckets     =         2048]
----------------------------------------------------------
  3000 requests in 10.00s, 785.16KB read
  Non-2xx or 3xx responses: 3000
Requests/sec:    299.99
Transfer/sec:     78.51KB
```

**Comment by cdelimitrou** [comment]
It looks like it completed the number of requests you specified. The achieved requests per sec are 299.99. Did you see any errors?

**Comment by mrbeann** [comment]
Yeah, all of them complete, but all failed. ` Non-2xx or 3xx responses: 3000`

**Comment by zyqCSL** [comment]
I would guess this is caused by the "localhost" in config.json. Grpc has some issue supporting loopback network, so in the Readme there is a note to replace "xxxx" with physical address. It seems the config.json is changed by a previous merge, with original "xxxx" replaced by localhost.

**Comment by mrbeann** [comment]
Yes, it seems to work now, but this workload seems only have four services?
![image](https://user-images.githubusercontent.com/9161747/70720772-67c73b80-1d2f-11ea-9b02-22c1b08ce095.png)

**Comment by Fatahir** [comment]
![wrk](https://user-images.githubusercontent.com/33696029/74087359-45426c00-4aad-11ea-9cac-48f73627b1a0.png)
i have the same issue. all the generated requests failed. how to solve this issue

**Comment by zyqCSL** [comment]
Services are listed in https://github.com/delimitrou/DeathStarBench/tree/master/hotelReservation/services. The current architecture can be described by the following figure 
![hotel_arch](https://user-images.githubusercontent.com/31255648/74668507-53486900-51e0-11ea-9088-2ab1a2582708.png)

**Comment by rukminiiyer** [comment]
This workload seems to have only four services? I am seeing only 4 services in Jaeger UI..

**Comment by vishnuvsingh** [comment]
I also have the same issue. I investigated this further by sending requests through postman, and for all the requests I am receiving a response of Internal Server Error (Status code 500). How can we resolve this?

**Comment by zyqCSL** [comment]
I tried running hotel reservation with our posted changes in https://github.com/delimitrou/DeathStarBench/issues/32#issuecomment-564782040 and wasn't able to reproduce the internal server error. 

I think your issue could be caused by 1) system environment configuration (use docker>=1.13), or more likely 2) misconfiguration of docker-compose file. Please take a look at https://docs.docker.com/compose/, understand how docker-compose works, and make sure configs in docker-compose.yml 
 meet your server configuration before running the benchmark.

**Comment by Fatahir** [comment]
@zyqCSL help me to solve this if you know.  /46[](https://github.com/delimitrou/DeathStarBench/issues/46)

**Comment by SuvigyaJain1** [comment]
I am having the same. Issue. On bare metal docker, the requests work without issue but when I deploy it to kubernetes I am getting 500 Internal Server Error for every request.

**Comment by denyshubh** [comment]
Hi, 
I am facing the same issue weather running using docker-compose or on kubernetes (gke, eks, rke2). I have tested the original code at [go-micro-service](https://github.com/harlow/go-micro-services) and seems to be working fine without any error. After looking at the logs, I am able to see that the **search** microservice is having some connectivity issue. 

Few of the Errors I was able to collect are
`
hotel_reserv_search       | WARNING: 2023/04/14 22:42:58 grpc: addrConn.createTransport failed to connect to {srv-geo.default.10.108.189.25.sslip.io:80 0  <nil>}. Err :connection error: desc = "transport: Error while dialing dial tcp 10.108.189.25:80: i/o timeout". Reconnecting...
` 

`
2023-04-14T22:34:37.869Z [WARN]  agent: [core][Channel #1 SubChannel #3] grpc: addrConn.createTransport failed to connect to {
consul_1                  |   "Addr": "dc1-127.0.0.1:8300",
consul_1                  |   "ServerName": "2968e4c5b5ee",
consul_1                  |   "Attributes": null,
consul_1                  |   "BalancerAttributes": null,
consul_1                  |   "Type": 0,
consul_1                  |   "Metadata": null
consul_1                  | }. Err: connection error: desc = "transport: Error while dialing dial tcp 127.0.0.1:0->127.0.0.1:8300: operation was canceled"`
